packages/widget/src/hooks/useExplorer.ts

```diff
@@ -0,0 +1,55 @@
+import type { Chain } from '@lifi/sdk';
+import { lifiExplorerUrl } from '../config/constants.js';
+import { useAvailableChains } from '../hooks/useAvailableChains.js';
+import { useWidgetConfig } from '../providers/WidgetProvider/WidgetProvider.js';
+
+const sanitiseBaseUrl = (baseUrl: string) => baseUrl.trim().replace(/\/+$/, '');
+
+export const useExplorer = () => {
+  const { explorerUrls } = useWidgetConfig();
+  const { getChainById } = useAvailableChains();
+
+  const getBaseUrl = (chain: Chain) => {
+    const baseUrl = explorerUrls?.[chain.id]
+      ? explorerUrls[chain.id][0]
+      : chain.metamask.blockExplorerUrls[0];
+
+    return sanitiseBaseUrl(baseUrl);
+  };
+
+  const getTransactionLink = (txHash: string, chain?: Chain) => {
+    if (!chain) {
+      const baseUrl = explorerUrls?.internal?.[0]
+        ? sanitiseBaseUrl(explorerUrls?.internal[0])
+        : lifiExplorerUrl;
+      return `${baseUrl}/tx/${txHash}`;
+    }
+
+    return `${getBaseUrl(chain)}/tx/${txHash}`;
+  };
+
+  const getTransactionLinkByChainId = (txHash: string, chainId: number) => {
+    return getTransactionLink(txHash, getChainById(chainId));
+  };
+
+  const getAddressLink = (address: string, chain?: Chain) => {
+    // TODO: consider here that LiFi explorer supports /wallet/ rather than /address/
```

Comment:
Lifi Explorer (and Jumper Scan too) `/wallet/` gets used in place of  `/address/`

https://github.com/lifinance/lifi-explorer/blob/develop/packages/explorer/src/config/routes.ts

In the current WIP version I have set this up so you can see the problem.

1. Connect your wallet
2. Select Arbitrum as your chain in the wallet
3. And then click the explorer link from the widget wallet menu

Should we change the widget configuration on this to allow the tx and address paths to be overridable?

something like this?

```
explorerUrls: {
    42161: {
      baseUrl: 'https://jumper.exchange/scan',
    },
    internal: {
      baseUrl: 'https://jumper.exchange/scan',
      addressPath: '/wallet/',
      txPath: '/transaction/',
   },
}
```

Reply:
This is a possible option in future should we need it. Staying with the current structure for now